### PR TITLE
DOC: add example for plotting asymmetrical error bars

### LINKED ISSUE #41034: DOC: there is no example for plotting asymmetrical errorbars
#### Location of the documentation

[errorbar visualization](https://pandas.pydata.org/pandas-docs/stable/user_guide/visualization.html#plotting-with-error-bars)

#### Documentation problem

Currently there's only an example of the symmetrical case, i.e. where the lower/upper errors are the same. The asymmetrical case is supported but actually quite tricky to get to work. I could only find one [stackoverflow post](https://stackoverflow.com/questions/45798492/asymmetrical-errorbar-with-pandas) and it is not especially helpful

#### Suggested fix for documentation

I suggest adding an example that showcases how to plot asymmetrical errorbars

### PR DESCRIPTION
**Pull request by mortada**
- [x] closes #41034 
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/user_guide/visualization.rst [hunk 1]
```diff
@@ -1496,6 +1494,30 @@ Here is an example of one way to easily plot group means with standard deviation
...
+Asymmetrical error bars are also supported, however raw error values must be provided in this case. For a ``N`` length :class:`Series`, a ``2xN`` array should be provided indicating lower and upper (or left and right) errors. For a ``MxN`` :class:`DataFrame`, asymmetrical errors should be in a ``Mx2xN`` array.
+
+Here is an example of one way to plot the min/max range using asymmetrical error bars
+
+.. ipython:: python
+
+   mins = gp3.min()
+   maxs = gp3.max()
+   mins
+   maxs
```
Comment (rhshadrach) [inline doc/source/user_guide/visualization.rst:1506]: It doesn't seem to me like these values need to be displayed. Do they make the example more clear?

File: doc/source/user_guide/visualization.rst [hunk 2]
```diff
@@ -1496,6 +1494,30 @@ Here is an example of one way to easily plot group means with standard deviation
 
    plt.close("all")
 
+Asymmetrical error bars are also supported, however raw error values must be provided in this case. For a ``N`` length :class:`Series`, a ``2xN`` array should be provided indicating lower and upper (or left and right) errors. For a ``MxN`` :class:`DataFrame`, asymmetrical errors should be in a ``Mx2xN`` array.
+
+Here is an example of one way to plot the min/max range using asymmetrical error bars
```
Comment (rhshadrach) [inline doc/source/user_guide/visualization.rst:1499]: nit: Period at the end.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: doc/source/user_guide/visualization.rst (hunk 1 above)
Comment (mortada) [reply doc/source/user_guide/visualization.rst]: indeed not really necessary, removed

File: doc/source/user_guide/visualization.rst (hunk 2 above)
Comment (mortada) [reply doc/source/user_guide/visualization.rst]: added

**Comment by mortada** [comment]
@rhshadrach PR updated, thanks!

**Review by jreback — APPROVED** [review]
lgtm. @rhshadrach over to you

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @mortada!

**Pull request merged**